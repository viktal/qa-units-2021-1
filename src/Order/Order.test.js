import React from 'react'

import {configure, shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Order from "./Order";
import {fakeOrders} from "../data/fakeOrders";

jest.mock('../utils/getDate')


const fakeDate = 123;
import {getDate} from '../utils/getDate';

configure({adapter: new Adapter()});


describe('Order.js', () => {

    beforeEach(() => {
        getDate.mockReturnValue(fakeDate);
    });

    afterEach(() => {
        jest.resetModules();
    });

    it('render empty', () => {
        const wrapper = shallow(<Order/>);
        expect(wrapper).toEqual({});
    });

    it('render with data', () => {
        const wrapper = shallow(<Order order={{shop: "test", date: 1}}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render without params', () => {
        const wrapper = shallow(<Order order={{}}/>);
        expect(wrapper).toEqual({});
    });

    it('render with data and items', () => {
        const wrapper = shallow(<Order order={{shop: "test", date: 1, items: ["aa", "ss", "dd"]}}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('check call getDate', () => {
        shallow(<Order order={fakeOrders[0]}/>);
        expect(getDate).toHaveBeenCalled();
    });
});
