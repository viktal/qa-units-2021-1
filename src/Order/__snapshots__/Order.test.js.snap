// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render with data 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      test
    </span>
    <span
      id="date"
    >
      123
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js render with data and items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      test
    </span>
    <span
      id="date"
    >
      123
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      aa
    </div>
    <div
      className="Order-item"
      key="1"
    >
      ss
    </div>
    <div
      className="Order-item"
      key="2"
    >
      dd
    </div>
  </div>
</div>
`;
